contracts/crowdsale/validation/CappedCrowdsale.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+
+/**
+ * @dev Crowdsale with a limit for total contributions
+ */
+contract CappedCrowdsale is Crowdsale {
+  using SafeMath for uint256;
+
+  uint256 public cap;
+
+  /**
+   * @param _cap Max amount of wei to be contributed
+   */
+  function CappedCrowdsale(uint256 _cap) public {
+    require(_cap > 0);
+    cap = _cap;
+  }
+
+  /**
+   * @return Whether the cap was reached
+   */
+  function capReached() public view returns (bool) {
+    return weiRaised >= cap;
+  }
+
+  function _postValidatePurchase(address _beneficiary, uint256 _weiAmount) internal {
+    super._postValidatePurchase(_beneficiary, _weiAmount);
+    require(weiRaised <= cap);
+  }
```

Comment:
What about turning this check into `require(weiRaised + msg.value <= cap)` in `_preValidatePurchase` and save some gas for the poor guy who was left out the crowdsale?